aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -785,15 +792,48 @@ internal class RealAWSCognitoAuthPlugin(
         onSuccess: Consumer<AuthUpdateAttributeResult>,
         onError: Consumer<AuthException>
     ) {
-        TODO("Not yet implemented")
+        authStateMachine.getCurrentState { authState ->
+            when (authState.authNState) {
+                // Check if user signed in
+                is AuthenticationState.SignedIn -> {
+                    GlobalScope.launch {
+                        try {
+                            val attributes = listOf(attribute)
+                            val userAttributesOptions = options as? AWSCognitoAuthUpdateUserAttributesOptions
```

Comment:
I noticed `CAST NEVER SUCCEEDS` exception. Could you check if `AWSCognitoAuthUpdateUserAttributesOptions` extends AuthUpdateUserAttributeOptions

Reply:
Yes it does.